Try a different inbetweenserter approach

jasmussen opened the pull request:
This is an attempt to redesign the inbetweenserter to be less frustrating. Currently in master it's too easy to invoke it whe
n you don't mean to.

This branch changes that approach. You can still hover between two lines to reveal a line indicator. But you can't click this line. Instead, a new plus button sits at the center of this line, and you have to click that to insert between.

![inbetweenserter](https://user-images.githubusercontent.com/1204802/40228910-e30cb164-5a92-11e8-8c5b-62419f889bd3.gif)

Please give this a spin. Depending on feedback, it needs visual love, and other tweaks. Right now it's definitely a try branch, and needs your thoughts. For example, when you click to insert a provisional placeholder block, you're still hovering in an area where the inserter line will appear, but it should immediately fade out instead. We might also want to look at some of that sizing magic that @aduth used for the old inbetweenserter.

Commit:
Try a different inbetweenserter approach

This is an attempt to redesign the inbetweenserter to be less frustrating. Currently in master it's too easy to invoke it whe
n you don't mean to.

This branch changes that approach. You can still hover between two lines to reveal a line indicator. But you can't click this line. Instead, a new plus button sits at the center of this line, and you have to click that to insert between.

Please give this a spin. Depending on feedback, it needs visual love, and other tweaks. Right now it's definitely a try branch, and needs your thoughts.

chrisvanpatten commented:
Bugs aside, this feels like a big improvement. Almost every time I compose content with Gutenberg I accidentally create extra blocks when I click the inbetweenserter, and in my brief testing this really mitigates that! Nice work. Excited to see it polished up.

karmatosed reviewed (approved):
Whilst I know you're going to iterate I am approving just based on it being an improvement to what we have right now! Let's get this worked on and in as soon as possible.

jasmussen commented:
Yay! I see positive feedback and lots of hearts and approval. I'm going to do a polish pass on this, and see if we can fix the bugs.

youknowriad committed:
Hide inbetween inserter from empty paragraph blocks + css tweaks

jasmussen commented:
Rebased and squashed. Thanks to a contribution by Riad, it now looks like this:

![updated](https://user-images.githubusercontent.com/1204802/40410658-a9402dfa-5e6e-11e8-9938-9e19f84f0f6b.gif)

We should test that https://github.com/WordPress/gutenberg/pull/6818/files#diff-e0abfc1a3e466fbf4144905e29cf2ef5R22 doesn't cause any regressions. But otherwise, should we get this in early as Tammie suggests, and then look at separate iterations if need be?

afercia commented on editor/components/block-list/insertion-point.js:
```diff
@@ -26,18 +26,21 @@ class BlockInsertionPoint extends Component {
+							icon="insert"
+							className="editor-block-list__insertion-point-button"
+							onClick={ this.onClick }
+							aria-label={ __( 'Insert block' ) }
```
`aria-label`  doesn't work on the IconButton component, so this button is not labeled at all and announced just as "button". IconButton needs a `label` prop. That would also produce a tooltip, which is necessary to make the visually expose the name of the control.

Most importantly, the inserter doesn't show up on focus, when tabbing with the keyboard.

afercia commented:
Using a `label` prop instead of `aria-label` makes the button properly labeled. It also adds a tooltip:

<img width="730" alt="inserter" src="https://user-images.githubusercontent.com/1682452/40416488-b0de14fa-5e7d-11e8-853d-6d25fe3476bc.png">

Haven't looked in depth to check why the inserter doesn't appear when focused with the keyboard, but that should be fixed 🙂

youknowriad committed:
Use a labal prop instead of aria-label for the insertion point

youknowriad committed:
Make the inserter visible when focused

jasmussen commented:
Thanks, both Riad and Andrea! 🤘

youknowriad committed:
Fix e2e tests

youknowriad reviewed (approved):
LGTM 👍

jasmussen commented:
Let's try this!
